src/test/java/org/openmrs/contrib/qaframework/automation/ManageUserAccountSteps.java

```diff
@@ -0,0 +1,136 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import java.util.List;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.reference.page.ManageUserAccountPage;
+import org.openmrs.reference.page.SystemAdministrationPage;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class ManageUserAccountSteps extends Steps {
+
+	private static final String SYSTEM_ALERT = "Atleast 8 character(s) are required";
+	private SystemAdministrationPage systemAdministrationPage;
+	private ManageUserAccountPage userAccountPage;
+
+	@Before(RunTest.HOOK.SELENIUM_USER_ACCOUNT)
+	public void visitHomePage() {
+		initiateWithLogin();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_USER_ACCOUNT)
+	public void destroy() {
+		quit();
+	}
+
+	@Given("a user clicks on System Administartion app from home page")
+	public void loadSystemAdministrationPage() {
+		systemAdministrationPage = homePage.goToSystemAdministrationPage();
+	}
+
+	@Then("the system loads the system administration page")
+	public void systemloadsSystemAdministrationPage() {
+		assertTrue(textExists("System Administration"));
+	}
+
+	@When("a user clicks on Manage accounts app")
+	public void loadManageAccountsPage() {
+		userAccountPage = systemAdministrationPage.goToManageAccounts();
+	}
+
+	@Then("the system loads the manage acccount page")
+	public void systemloadsManageAccountsPage() {
+		assertTrue(textExists("Manage Accounts"));
+	}
+
+	@When("a user clicks Add New Account button")
+	public void goToAddNewAccountPage() {
+		userAccountPage.clickOnAddUser();
+	}
+
+	@Then("the system loads the add new account form")
+	public void systemLoadsAddNewAccountPage() {
+		assertTrue(textExists("Add New Account"));
+	}
+
+	@When("a user fills details of the Data clerk in the user account form")
+	public void registerDataClerkDetails() {
+		userAccountPage.enterPersonalDetails("Data", "Clerk");
+		userAccountPage.selectGender();
+		userAccountPage.clickOnAddUserAccount();
+		userAccountPage.enterUsername("dclerk");
```

Comment:
userAccountPage.enterUsername("data clerk"), would look so readable than dclerk

Reply:
> userAccountPage.enterUsername("data clerk"), would look so readable than dclerk

I read about a blog that explains best practices of creating `username` account though I can not retrieve the link to share it here. The write advises that a recommended username should be a single text not more than one text for example `dclerk` But not `data clerk`. OpenMRS geeks knew about this before that is why when one is logging into our demo server the username is either `admin` or `nurse` or `doctor`

Its from the reason above we embrace good coding practices in creating usernames as well. Any way this is just a test that runs in the background that we can set any username of our interest But we need to align to quality coding standards.

Reply:
> The write advises that a recommended username should be a single text not more than one text for example `dclerk` But not `data clerk`. OpenMRS geeks knew about this before that is why when one is logging into our demo server the username is either `admin` or `nurse` or `doctor`

Suppose anyone new to openmrs convention, sees dclerk as username, wouldnt it be looking so different from admin and nurse , i agree with single user names, would we then use something different like clerk or doctor or doctorclerk.

Reply:
> Suppose anyone new to openmrs convention, sees dclerk as username, wouldnt it be looking so different from admin and nurse

Being the geek behind the development of this test, I choose dclerk getting the first letter of Data and embedding it to the second name clerk coming up with dclerk. BTW this is just a test so as long as am in line with the coding standards all shouts out clearly well! From the actual ground the system admin will create a user name when creating user accounts that's stands out for him

Reply:
> BTW this is just a test so as long as am in line with the coding standards all shouts out clearly well!

Tests should be so clear to any implementor to bring out the aspect of quality assurance.